### PR TITLE
Add support for Kafka 3.4.0 and remove support for Kafka 3.2.x

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR adds support for Kafka 3.4.0 and removes support for Kafka 3.2.x (3.2.0, 3.2.1, 3.2.3). 

It also adds a document to developer docs which tries to describe how to add new Kafka versions which can be used as a checklist to make sure nothing is forgotten. It might not be perfect, but should help in the future releases and can continue refining it.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: development-docs/KAFKA_VERSIONS.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Adding / removing Kafka versions
+
+## Adding new Kafka versions
+
+There is no _definitive_ guide as every version might bring surprises.
+But these are some of the tasks you usually have to do.
+
+* If you add a new minor version, add new 3rd party libs directory to `docker-images/artifacts/kafka-thirdparty-libs`
+  * This directory is typically shared by all patch releases which are part of the same minor release. So it is named `X.Y.x` - e.g. `1.1.x`
```
Comment (ppatierno) [inline development-docs/KAFKA_VERSIONS.md:9]: why not x.y.z? or X.Y.z?

File: development-docs/KAFKA_VERSIONS.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  * Add a `pom.xml` file with dependencies and make sure they are aligned with the dependencies inside Kafka (e.g. use similar version of Jackson libraries)
+  * (Typically, you would do this by copying the directory for the previous minor version)
```
Comment (ppatierno) [inline development-docs/KAFKA_VERSIONS.md:11]: why this is in parenthesis? it's weird to see a topic on a list in parenthesis, imho.

File: development-docs/KAFKA_VERSIONS.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* Add the new version to `kafka-versions.yaml` file
+  * If the version you are adding is the latest, mark it as default and unmark the previous one to not be default anymore
```
Comment (ppatierno) [inline development-docs/KAFKA_VERSIONS.md:13]: maybe more explicitly "mark it as `default: true` and unmark (as `default: false`) the previous ..." or something similar?

File: development-docs/KAFKA_VERSIONS.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  * Update the main `pom.xml` to use the latest Kafka version in the operator
+  * Update `systemtest/src/test/resources/upgrade/StrimziUpgradeST.yaml` with the new version
+  * Update protocol versions in `cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConfigurationTests.java` if needed
+* Make sure the `cluster-operator/src/test/java/io/strimzi/operator/cluster/KafkaVersionTestUtils.java` is up-to-date
+* If you are adding a release candidate which is not yet published to Maven central, make sure to update the `ST_FILE_PLUGIN_URL_DEFAULT` field in `systemtest/src/main/java/io/strimzi/systemtest/Environment.java` to get the system tests to pass.
+* Run `make all` to update all the installation files, generated docs files, Helm chart etc.
+  You will need to commit also all the files it updates before opening a PR.
+* Add a `CHANGELOG.md` record
+* Run unit tests
+* Run system tests
```
Comment (ppatierno) [inline development-docs/KAFKA_VERSIONS.md:28]: you can run this after opening the PR as "regression" on Azure, right?

File: development-docs/KAFKA_VERSIONS.md (hunk 1 above)
Comment (scholzj) [reply development-docs/KAFKA_VERSIONS.md]: I renamed it to `<MAJOR>.<MINOR>.x`. Hope that helps.

Comment (ppatierno) [reply development-docs/KAFKA_VERSIONS.md]: Yeah much better, thanks!

**Review by ppatierno — APPROVED** [review]

**Pull request merged**